problemtools/verifyproblem.py

```diff
@@ -740,6 +741,296 @@ def check(self, args):
+                if j == 0:
+                    new += group
+                else:
+                    group, rest = group.split('}')
+                    if group.startswith('seed'):
+                        new += seed
+                    elif group == 'name':
+                        new += name
+                    else:
+                        return err()
+                    new += rest
+            parts[i] = new
+
+        program, arguments = parts[0], parts[1:]
+        if program not in self._generators:
+            self._generators[program] = program
+
+        return (program, arguments)
+
+    def _parse_testcase(self, data, state):
+        if state['input'] is None:
+            self.error('Path %s in generators.yaml must contain an input key' % state['path'])
+        for key in ['input', 'solution', 'visualizer']:
+            if state[key] is not None:
+                state[key] = self._parse_command(key, state)
+        if state['input'] is not None:
+            self._testcases.append(state)
+
+    def _parse_directory(self, data, state):
+        # TODO: Process includes
+
+        if 'testdata.yaml' in data:
+            content = data['testdata.yaml']
+            if content is None:
+                content = {}
+            self._testdata_yaml['%s/%s' % (state['path'], 'testdata.yaml')] = content
+
+        cases = data.get('data', {})
+        ordered = True
+        if not isinstance(cases, list):
+            ordered = False
+            cases = [cases]
+
+        case_counter = 0
+        case_format = '%%0%dd' % len(str(len(cases)))
+        for case in cases:
+            if not isinstance(case, dict):
+                self.error('Path %s/data in generators.yaml must contain a dict or a list of dicts' % state['path'])
+                continue
+
+            if ordered:
+                case_counter += 1
+
+            for name, value in sorted(case.items(), key=lambda kv: str(kv[0])):
+                if ordered:
+                    num = case_format % case_counter
+                    name = num + ('' if name is None else '-' + str(name))
+                else:
+                    name = str(name)
+
+                next_state = copy.deepcopy(state)
+                next_state['path'] = '%s/%s' % (state['path'], name)
+                self._parse_element(value, next_state)
+
+    def _parse_element(self, data, state):
+        if sys.version_info[0] < 3 and data == '':
+            data = None
+        if data is None:
+            data = '/%s.in' % state['path']
+            state['manual'] = True
+        if isinstance(data, str):
+            data = { 'input': data }
+        if not isinstance(data, dict):
+            self.error("Path %s in generators.yaml must specify a dict" % state['path'])
+            return
+
+        state.update({
+            key: data[key]
+            for key in Generators._TESTCASE_OPTIONS
+            if key in data
+        })
+        if sys.version_info[0] < 3:
```

Comment:
Can you add a comment what this if is doing?

Reply:
This was accounting for some odd behavior in the python2 yaml library, but it seems like it has been fixed as I cannot reproduce it anymore. Removed.